### ISSUE TITLE
SplitFactory does not parse settings.core.authorizationKey to String

### ISSUE THREAD

**Issue by rarenatoe** [splitio/react-native-client#19]
SplitFactory does not parse `settings.core.key` into a String value, which causes errors when something like a number is passed.
This error does not happen with `@splitsoftware/splitio`

Example:
<img width="433" alt="Screen Shot 2022-03-24 at 15 43 51" src="https://user-images.githubusercontent.com/44821537/160007038-02849cd8-df1c-4f94-a8e8-3363c0d9c4bc.png">

**Comment by NicoZelaya** [comment]
Hi @rarenatoe ! 

The SDK always expects strings so that's what the input validation will consider valid, but we do log a warning and convert on the `splitio` package and this one is expected to do the same. 
I'll open up a ticket to review this one in the upcoming weeks but in the meantime, do you think you can progress by providing a string as expected? 
One possible option that should work for strings and numbers is to convert it by concatenating an empty string to your number, and wouldn't modify if you send the expected type of string as well.

```
// Instantiate the SDK
const factory: SplitIO.ISDK = SplitFactory({ 
  core: {
    authorizationKey: 'YOUR_API_KEY',
    key: something.userID + ''   // where something.userID represents your dynamic variable with that information.
  }
});
```

Hopefully the workaround works but if you need further assistance, I suggest to communicate through support@split.io to share more details from your goals and implementation. 

Best,
Nico.

**Comment by rarenatoe** [comment]
Yes, no worries. I got it fixed on my side. I just thought you guys would want to know of this irregular behavior between the two repositories.

**Comment by EmilianoSanchez** [comment]
Hi @rarenatoe , 

We have published v0.2.0 of the SDK (https://www.npmjs.com/package/@splitsoftware/splitio-react-native), that includes a fix for this issue to behave as `@splitsoftware/splitio`, among other updates.

I will close the issue, but feel free to reopen it if you still find the error on your side.
